test/bats/e2e-provider.bats

```diff
@@ -66,54 +67,86 @@ SLEEP_TIME=1
-  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+  kubectl delete -f $BATS_RESOURCE_YAML_DIR/service_account_token_secretsync.yaml
+}
 
-  # Check the status of SecretSync in ss-namespace
-  status=$(kubectl get secretsyncs.secret-sync.x-k8s.io/sse2esecret -n ss-namespace -o jsonpath='{.status.conditions[0]}')
+@test "Cannot create a secret with invalid annotations" {
+  expected_message="The secretsyncs \"sse2einvalidannotationssecret\" is invalid: : ValidatingAdmissionPolicy 'secrets-store-sync-controller-validate-annotation-policy' with binding 'secrets-store-sync-controller-validate-annotation-policy-binding' denied request: One of the annotations applied on the secret has an invalid format. Update the annotation and try again."
 
-  expected_message="Secret update failed because the controller could not retrieve the Secret Provider Class or the SPC is misconfigured. Check the logs or the events for more information."
-  expected_reason="ControllerSPCError"
-  expected_status="False"
+  deploy_spc_ss_expect_failure \
+    "$BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml" \
+    "$BATS_RESOURCE_YAML_DIR/invalid_annotation_key_secretsync.yaml" \
+    "sse2einvalidannotationssecret" \
+    "$expected_message"
 
-  # Extract individual fields from the status
-  message=$(echo $status | jq -r .message)
-  reason=$(echo $status | jq -r .reason)
-  status_value=$(echo $status | jq -r .status)
+  deploy_spc_ss_expect_failure \
+    "$BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml" \
+    "$BATS_RESOURCE_YAML_DIR/invalid_annotation_value_secretsync.yaml" \
+    "sse2einvalidannotationssecret" \
+    "$expected_message"
+}
 
-  # Verify the status fields
-  [ "$message" = "$expected_message" ]
-  [ "$reason" = "$expected_reason" ]
-  [ "$status_value" = "$expected_status" ]
+@test "Cannot create a secret with invalid labels" {
+  expected_message="The secretsyncs \"sse2einvalidlabelsecret\" is invalid: : ValidatingAdmissionPolicy 'secrets-store-sync-controller-validate-label-policy' with binding 'secrets-store-sync-controller-validate-label-policy-binding' denied request: One of the labels applied on the secret has an invalid format. Update the label and try again."
 
-  # Check that the secret is not created in ss-namespace
-  cmd="kubectl get secret sse2esecret -n ss-namespace"
-  run $cmd
-  assert_failure
+  deploy_spc_ss_expect_failure \
+    "$BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml" \
+    "$BATS_RESOURCE_YAML_DIR/invalid_label_key_secretsync.yaml" \
+    "sse2einvalidlabelsecret" \
+    "$expected_message"
+
+  deploy_spc_ss_expect_failure \
+    "$BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml" \
+    "$BATS_RESOURCE_YAML_DIR/invalid_label_value_secretsync.yaml" \
+    "sse2einvalidlabelsecret" \
+    "$expected_message"
 }
 
 teardown_file() {
   archive_provider "app=secrets-store-sync-controller" || true
   archive_info || true
 
   if [[ "${INPLACE_UPGRADE_TEST}" != "true" ]]; then
```

Comment:
Remove this check because we don't have those tests yet. As part of every e2e run, let's just start with cleaning up everything and can add upgrade tests/check after.